src/cdomains/intDomain.ml

```diff
@@ -923,6 +923,461 @@ struct
   let project ik p t = t
 end
 
+module IntervalSetFunctor(Ints_t : IntOps.IntOps): S with type int_t = Ints_t.t and type t = (Ints_t.t * Ints_t.t) list =
+struct
+
+  module Interval_functor = IntervalFunctor(Ints_t) 
+
+  let name () = "IntervalSets"
+
+  type int_t = Ints_t.t
+
+  type t = (Ints_t.t * Ints_t.t) list [@@deriving eq, hash, ord]
```

Comment:
Might be worth commenting here that these are given in ascending order and are free of overlaps.